tool/tctl/common/resource_command.go

```diff
@@ -89,7 +90,7 @@ func (rc *ResourceCommand) Initialize(app *kingpin.Application, config *service.
 	rc.config = config
 
 	rc.createCmd = app.Command("create", "Create or update a Teleport resource from a YAML file")
-	rc.createCmd.Arg("filename", "resource definition file").Required().StringVar(&rc.filename)
+	rc.createCmd.Arg("filename", "resource definition file, empty for stdin").StringVar(&rc.filename)
```

Comment:
Could we maybe make this more like other Unix tools where `-` is used to refer to stdin? At the moment, if you `tctl create -f` then it'll just hang there waiting for input which might be confusing. I think `kubectl` does the same thing i.e. `echo "wordy yaml" | kubectl create -f -`

Reply:
Difference with kubectl is that it uses `-f` to refer to the filename and it simply takes `-` as a value, whilte `tctl` uses `-f` for another purpose so to copy kubectl's behavior we might have to introduce another flag instead of the filename being a (positional) argument:
```
tctl create -f -n=-
```
Note the need to use `=` to separate the value - it's a requirement in kingpin's argument parser.
If the filename were to continue being a positional argument, it'd result in the following error with the version of kingpin we're using:
```
tctl create -f -
error: unknown short flag '-'
```
The problem is that it reserves `-` for short flags and expects a non-empty flag following a dash (actually it does not enforce it but still parses this as a short flag resulting in the error). It could be thought that we could solve this by turning off [flag interspersion](https://godoc.org/github.com/gravitational/kingpin#Application.Interspersed). Unfortunately:
  * interspersion is top-level only (e.g. one can only configure it on the top-level Application)
  * flags are still parsed first

I see a couple of ways to work this around:
  * re-purpose -f (--filename) as a flag and rename force to e.g. `overwrite` (risky as not backwards-compatible). Will still require tweaking the code of kingpin if one wants `-f -` notation (w/o `=`)
  * turn `filename` into a flag as above under a new short name and settle on `tctl create -f -n=-` as a notation for stdin
  * patch kingpin to properly handle orphaned `-` as an argument. I already attempted to solve a similar issue when I was introducing support for flag interspersion for sub-commands [here](https://github.com/gravitational/kingpin/pull/3) which would require a bit of improvement.

WDYT?

Reply:
Most unix tools (cat, tail, wc, grep, ...) read from stdin when no filename is provided.
I think the current implementation is appropriate.

Reply:
I'd forgotten that `-f` is actually `--force` rather than `--file`. With that in mind I think just reading from stdin when there's no name provided is fine, although I will inevitably forget this behaviour many times in the future!